packages/flutter/lib/src/material/tabs.dart

```diff
@@ -982,11 +982,11 @@ class _TabBarState extends State<TabBar> {
       _initIndicatorPainter();
     }
 
-    if (widget.tabs.length > oldWidget.tabs.length) {
-      final int delta = widget.tabs.length - oldWidget.tabs.length;
+    if (widget.tabs.length > _tabKeys.length) {
```

Comment:
Can we not consider both cases here?
The test you deleted is a valid use case that we fixed in the past. Folks expecting that behavior will be broken by this change.

Reply:
The test I deleted passes even after my change. If it may cause an issue later, I'll add it back in.

Reply:
YEs, please do add it back in. We want to make sure we don't regress. More tests are never a bad thing. :)